micro-rdk-installer/src/main.rs

```diff
@@ -331,3 +367,105 @@ fn main() -> Result<(), Error> {
     };
     Ok(())
 }
+
+fn update_app_image(args: &AppImageArgs) -> Result<(), Error> {
+    let config = Config::load().map_err(|err| Error::SerialConfigError(err.to_string()))?;
+
+    let dir = tempfile::tempdir().map_err(Error::FileError)?;
+    let tmp_old = dir.path().join("running-ptable.img");
+
+    log::info!("Retrieving running partition table");
+    let mut flasher =
+        connect(&args.connect_args, &config, false, false).map_err(|_| Error::FlashConnect)?;
+    flasher
+        .read_flash(
+            PARTITION_TABLE_ADDR.into(),
+            PARTITION_TABLE_SIZE.into(),
+            0x1000,
+            64,
+            tmp_old.clone(),
+        )
+        .map_err(|_| Error::FlashConnect)?;
+
+    let old_ptable_buf = fs::read(tmp_old).map_err(Error::FileError)?;
+    let old_ptable = PartitionTable::try_from_bytes(old_ptable_buf.clone())
+        .map_err(|e| Error::PartitionTableError(e.to_string()))?;
+
+    log::info!("Retrieving new image");
+    let tmp_new = tempfile::Builder::new()
+        .prefix("micro-rdk-bin")
+        .tempdir()
+        .map_err(Error::FileError)?;
+    let app_path_new = match &args.flash_args.bootloader {
```

Comment:
Why `bootloader` here? Not saying that it is wrong, just a little surprising.

Reply:
It's just how espflash's clap structs have things named. Not a huge fan of it

Reply:
renamed it to `binary_path`